test/Microsoft.ServiceFabric.Diagnostics.Tests/Metrics/Int64Meter1DTest.cs

```diff
@@ -0,0 +1,74 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Fuzzy;
+using Inspector;
+using Microsoft.ServiceFabric.Diagnostics.Metrics.Implementation;
+using Moq;
+using Xunit;
+
+namespace Microsoft.ServiceFabric.Diagnostics.Metrics
+{
+    public abstract class Int64Meter1DTest
+    {
+        readonly IMeter1D<long> sut;
+
+        // Constructor parameters
+        readonly IFabricMeter fabricMeter = Mock.Of<IFabricMeter>();
+        readonly List<string> systemDimensions = fuzzy.List(() => fuzzy.String(), Count.Min(1));
+
+        // Test fixture
+        static readonly IFuzz fuzzy = new RandomFuzz(Environment.TickCount);
+
+        public Int64Meter1DTest() => sut = new Int64Meter1D(fabricMeter, systemDimensions);
+
+        public class Class : Int64Meter1DTest
+        {
+            [Fact]
+            public void InvokesBaseWithGivenArguments()
+            {
+                var meter = (Meter)sut;
+                Assert.Same(fabricMeter, meter.Field<IFabricMeter>().Value);
+                Assert.Same(systemDimensions, meter.Field<IEnumerable<string>>().Value);
+            }
+        }
+
+        public class Record : Int64Meter1DTest
+        {
+            readonly string customDimension1 = fuzzy.String();
+            readonly long value = fuzzy.Int64();
+
+            [Fact]
+            public void CallsFabricMeterRecordWithMultipleSystemDimensions()
+            {
+                // Arrange
+                string[] expectedArray = systemDimensions.Concat(new[] { customDimension1 }).ToArray();
+
+                // Act
+                sut.Record(value, customDimension1);
+
+                // Assert
+                Mock.Get(fabricMeter).Verify(m => m.Record(value, (uint)expectedArray.Length, It.Is<string[]>(arr => arr.SequenceEqual(expectedArray))), Times.Once);
+            }
+
+            [Fact]
+            public void CallsFabricMeterRecordWithNoSystemDimensions()
```

Comment:
Can this test ever fail independently of the other? Ideally, each test should fail independently and pinpoint the problem with the information in its name. So, if it doesn't fail independently, it is likely redundant and would be best to remove. This goes against the common test-later practices when we aren't sure which tests are needed and which aren't. In this case, however, the system under test is small and well defined, so we should be able to verify if this test is needed by trying to sabotage the product code. If we can't modify the product code in a way that makes this test fail without the other, it's redundant.